plugin.picture.googlephotos/main.py

```diff
@@ -0,0 +1,395 @@
+        media = get_items(pageToken)
+        if not media:
+            return None
+        utils.storeData(path, media)
+
+
+def list_media():
+    # For all photo directories
+
+    path = profile_path + config.media_filename
+    if not args.get('call_type'):
+        xbmcvfs.delete(path)
+
+    result = utils.loadData(path)
+    if not result:
+        media = get_items()
+        if media:
+            utils.storeData(path, media)
+            result = [media]
+        else:
+            xbmc.executebuiltin(
+                f'Notification({__addon__.getLocalizedString(30413)}, time=3000)')
+            return
+    # List for media
+    items = []
+    for item in result:
+        items += create_media_list(item)
+
+    # Add list to directory
+    xbmcplugin.addDirectoryItems(
+        addon_handle, items, totalItems=len(items))
+
+    if 'nextPageToken' in result[-1]:
+        url = utils.build_url(base_url, {'mode': 'refresh', 'prev_q': qs})
+        li = xbmcgui.ListItem(__addon__.getLocalizedString(30414))
+        li.setProperty('IsPlayable', 'false')
+        xbmcplugin.addDirectoryItem(addon_handle, url, li)
+    if args.get('call_type'):
+        updateListing = True
+    else:
+        updateListing = False
+    xbmcplugin.endOfDirectory(
+        addon_handle, updateListing=updateListing, cacheToDisc=False)
+
+    if 'nextPageToken' in result[-1]:
+        loader = threading.Thread(target=get_items_bg, args=(
+            result, path,))
+        loader.start()
+        loader.join()
+
+
+def create_media_list(media: dict):
+    # Creates a list of (url, li) from media dictionary
+    items = []
+    for item in media.get("mediaItems", {}):
+        item_type = item["mimeType"].split('/')[0]
+        li = xbmcgui.ListItem(item["filename"])
+        url = None
+        h = xbmcgui.getScreenHeight()
+        w = xbmcgui.getScreenWidth()
+        if item_type == 'image':
+            # thumb_url = item["baseUrl"] + f'=w{w}-h{h}'
+            thumb_url = item["baseUrl"] + f'=w{960}-h{540}'
+            img_url = item["baseUrl"] + f'=w{w}-h{h}'
+            li.setArt(
+                {'icon': 'DefaultIconInfo.png'})
+            url = img_url
+        elif item_type == 'video':
+            vid_url = item["baseUrl"] + '=dv'
+            thumb_url = item["baseUrl"] + f'=w{w}-h{h}'
+            url = utils.build_url(
+                base_url, {'mode': 'play_video', 'path': vid_url,
+                           'status': item["mediaMetadata"]["video"]["status"], 'token_filename': token_path.name})
+            li.setArt({'thumb': thumb_url, 'icon': thumb_url})
+            li.setProperty('IsPlayable', 'true')
+        else:
+            continue
+        li.setProperty('MimeType', item["mimeType"])
+        items.append((url, li))
+    return items
+
+
+def play_video():
+    # https://kodi.wiki/view/HOW-TO:Video_addon
+    if args.get('status')[0] != 'READY':
+        xbmc.executebuiltin(
```

Comment:
Use `xbmcgui` methods instead of `xbmc.executebuilti()`

Reply:
Done